## Rubric

Instructions: DELETE this cell before you submit via a `git push` to your repo before deadline. This cell is for your reference only and is not needed in your report. 

Scoring: Out of 10 points

- Each Developing  => -2 pts
- Each Unsatisfactory/Missing => -4 pts
  - until the score is 

If students address the detailed feedback in a future checkpoint they will earn these points back


|                  | Unsatisfactory                                                                                                                                                                                                    | Developing                                                                                                                                                                                              | Proficient                                     | Excellent                                                                                                                              |
|------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|
| Data relevance   | Did not have data relevant to their question. Or the datasets don't work together because there is no way to line them up against each other. If there are multiple datasets, most of them have this trouble | Data was only tangentially relevant to the question or a bad proxy for the question. If there are multiple datasets, some of them may be irrelevant or can't be easily combined.                       | All data sources are relevant to the question. | Multiple data sources for each aspect of the project. It's clear how the data supports the needs of the project.                         |
| Data description | Dataset or its cleaning procedures are not described. If there are multiple datasets, most have this trouble                                                                                              | Data was not fully described. If there are multiple datasets, some of them are not fully described                                                                                                      | Data was fully described                       | The details of the data descriptions and perhaps some very basic EDA also make it clear how the data supports the needs of the project. |
| Data wrangling   | Did not obtain data. They did not clean/tidy the data they obtained.  If there are multiple datasets, most have this trouble                                                                                 | Data was partially cleaned or tidied. Perhaps you struggled to verify that the data was clean because they did not present it well. If there are multiple datasets, some have this trouble | The data is cleaned and tidied.                | The data is spotless and they used tools to visualize the data cleanliness and you were convinced at first glance                      |


# COGS 108 - Data Checkpoint

## Authors

Instructions: REPLACE the contents of this cell with your team list and their contributions. Note that this will change over the course of the checkpoints

This is a modified [CRediT taxonomy of contributions](https://credit.niso.org). For each group member please list how they contributed to this project using these terms:
> Analysis, Background research, Conceptualization, Data curation, Experimental investigation, Methodology, Project administration, Software, Visualization, Writing – original draft, Writing – review & editing

Example team list and credits:
- Alice Anderson: Conceptualization, Data curation, Methodology, Writing - original draft
- Bob Barker:  Analysis, Software, Visualization
- Charlie Chang: Project administration, Software, Writing - review & editing
- Dani Delgado: Analysis, Background research, Visualization, Writing - original draft

## Research Question

How do seasonal (Winter:Dec-Feb, Spring:Mar-May, Summer:Jun-Aug, Fall:Sep-Nov) differences influence recreation visitation and average recreation hours per visit (Recreation hour/Recreation visits) between U.S. National Parks in the Northeast, Midwest, Intermountains, and Pacific West regions from 2000 to 2024?



## Background and Prior Work

As a team, the famous U.S National Parks grabbed our attention during the process of selecting a topic. We believed that there were many interesting datasets and analysis that could be made regarding this topic. The U.S National Parks are spread among different states. However, they are different in many aspects. For instance, some national parks are more populated and easily accessible, while some are in less populated areas and are harder to get to. Each region's national park has its own unique features. Generally speaking, National Parks that are located near the major cities will be more accessible than the ones that are located in the more remote areas, which the latter will require greater effort and planning to visit. 

In addition to the regional difference, National Parks also have seasonal differences. Each national park has its own peak and off-season based on what tourists plan to do there. These factors greatly influence tourists in deciding whether they choose to visit which national park and the duration of their stay there. Hence, this project aims to explore how regional and seasonal differences in the US National Parks from 2000 to 2024 affect recreational visitation and hours spent per visit. 

A report by the National Park Service (NPS) shows that the reported number of recreational visits is 331.9 million in the year 2024, with visitors spending approximately 1.4 billion hours in national parks.<a name="ref1"></a>[1](#footnote1) Additionally, visitors distributed their visits throughout the year, where more than half of the parks show peak periods in February to June, and October to December. This shows that the peak season is during spring and fall which is the opposite to popular belief that the peak season is summertime. Instead, tourists visit throughout the year in different periods, and over half of the park shows that spring and fall had an above-average number of visits.

The National Park Service also found out that climate change not only affects resources of national parks, but they discovered that it influenced visitation patterns. Their data shows that 95% of parks has great relationship between air temperature and visitation.<a name="ref2"></a>[2](#footnote2) Visitation declines as the monthly average temperature is over 80F. This shows that a lot of the park visitors avoid hot weather, which leads to this phenomenon.

A report from the NPS in 2021 shows that the most visited national park is the Great Smoky Mountain National Park with 14 million recreation visits, with Zion National Park in second with 5 million recreation visits.<a name="ref3"></a>[3](#footnote3) The main reason why the Great Smoky Mountain National Park is so popular is due to its geographical location. It is located within a day's drive for most US population including cities like Chicago, Atlanta, Nashville, etc. Which made it very easy and accessible for a wide range of the population to visit without having to drive too far.<a name="ref4"></a>[4](#footnote4)


<a name="footnote1"></a>1. [^](#ref1) National Park Service. “VUStats Dashboard - Social Science (U.S. National Park Service).” *Www.nps.gov*, 17 Oct. 2024, www.nps.gov/subjects/socialscience/visitor-use-statistics-dashboard.htm.

<a name="footnote2"></a>
2. [^](#ref2) Fisichelli, Nicholas A., et al. “Protected Area Tourism in a Changing Climate: Will Visitation at US National Parks Warm up or Overheat?” *PLOS ONE*, vol. 10, no. 6, 17 June 2015, p. e0128226, https://doi.org/10.1371/journal.pone.0128226.

<a name="footnote3"></a>
3. [^](#ref3) “Visitation Highlights - Social Science (U.S. National Park Service).” *Nps.gov*, 2018, www.nps.gov/subjects/socialscience/highlights.htm.

<a name="footnote4"></a>
4. [^](#ref4) Gatlinburg, Mailing Address: 107 Park Headquarters Road, and TN 37738 Phone:436-1200 Contact Us. “Visitor Experience & Stewardship - Great Smoky Mountains National Park (U.S. National Park Service).” *Www.nps.gov*, www.nps.gov/grsm/learn/management/ves.htm.


## Hypothesis


We predict that the winter season (defined as December throught Feb) will have lower recreation hours and recreation visits across all parks, due to colder weather and shorter daylight hours. We also predict that this effect will be stronger in the Intermountains region due to average higher altitudes and snow. Using data about total recreation hours and total recreation visits, we also believe that the average time of recreation visits may be lower during winter months, which could be related to hours of daylight. 

(and at more northern regions, recreation hours and visitation frequency will decrease. We believe that this is potentially due to increased weather restrictions for snow, shorter daylight hours, and less people wanting to be in the outdoors for extended periods of time due to chilly temperatures. Therefore we predict that there will be a strong correlation between season of the year, and recreation hours, as well as season of the year and visitation frequency)

## Data

### Data overview
For each dataset include the following information
- Dataset #1
  - Dataset Name: National Park Service Visitor Use Statistics (IRMA)
  - Link to the dataset: https://irma.nps.gov/Stats/SSRSReports/National%20Reports/Query%20Builder%20for%20Public%20Use%20Statistics%20(1979%20-%20Last%20Calendar%20Year)
  - Number of observations:
    Monthy obersvations, for each park unit, for this project we will take data from (2000 -2024)
  - Number of variables:
     - National Parks
     - Year
     - Month
     - Recreation visits
     - Non-recreation visits
     - Recreation visitor hours
     - Non-recreation visitor hours
     - Overnight stays (by type) 
  - Recreation visits: are defiend as the number of people entering a park for recreational purposes in a given month. This will be used as the primary measure of park visitation
  - Visitor hours: used to construct a new variable. Average time spent per visit = recreation vistior hours/recreation visits. This will provides a proxy for visit length since the dataset does not directly report average duration
  - Year & Month: is to analyze trends over time, create season varaibles
     - Winter:Dec-Feb
     - Spring:Mar-May
     - Summer:Jun-Aug
     - Fall:Sep-Nov
  - Park Name used as a unique identifier to merge with external dadtasets (region, park type, weather)
  - Overnight stays:can be used additional measure of park use and to distinguish between day use and overnight visitation patterns
  - Non-recreation visits: will not be used in the main visitation measure because they include commuting and administrative use rather than tourism-related visits
  - Shortcomings is vistiations does not mean unique visitations, the dataset only takes into account of recreation visits not individual people. We cannot measure number of unique visitors and repeat visitation behavior.
  - There is no direction "average length of stay" because we only have recreation vistiro hours and recreation visits so we must construct average stay = vistiors hours/visits. This assumes that all visitor hours come form recreation visitors and reporitng is accruate accross parks
  - There is no visor characterisitcs, can not anaylze differences by age, income, residency (local vs toruosits), group size, transportation mode. So the analysis is limited to park patterns not people.
  - There is no weather in the dataset, you cannot directly measure temperature, rain, snow and extreme events, so you cannot test weather effects unless you merge naother dataset
  - Potential differences in counting methods across parks. They may use different tools like traffic counters, fee recipts, or manual estimates which can create inconsistency in visitation estimates especially for older data
  - Some missing or suppressed data: for certain praks or early years: some months may be missing, some overnight categories may be blank.
  - Measures park-level use, not specific activites: Based on the dataset we won't be able to tell from hiking vs camping vs sightseeing, or which areas insidee the park are crowded 


In [14]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [3]:
# Setup code -- this only needs to be run once after cloning the repo!
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
datafiles = [
    { 'url': 'https://drive.google.com/uc?export=download&id=1BanvjJ7btH6WlQlTZslI0XRIRQ8MDD8I', 'filename':'2000-2024 National Park Data.csv'}
]

get_data.get_raw(datafiles,destination_directory='data/00-raw/')

Overall Download Progress: 100%|██████████| 1/1 [00:01<00:00,  1.60s/it]                         


Successfully downloaded: 2000-2024 National Park Data.csv


### National Park Service Visitor Use Statistics Dataset

This data came from the Query Builder for Public Use Statistics provided by the National Parks Service. There are a few complications that must be considered when using this dataset. One is that  this dataset is that 2025 data is still preliminary, however this is not a problem for us since our cutoff for date is 2024. Additionally, some parks were established after 2004, so we must be conscious in our analyses that not all parks have data for every year. Not all parks report statistics, as they may be newer and lack the resources to collect such data. Some parks also shut down for COVID-19 and thus visitor data may be affected by this, which must be considered. The National Parks Service has had a formalized method of record keeping since the 1980s, which aims to collect statistics across parks in a way that allows for statistical analysis, and upkeeps quality. Therefore, there are no glaring issues regarding our dataset for this planned use.

Regions for our dataset include parks in the following states (Using state abbreviations):
- Northeast: ME, WV, VA 
- Intermountain: UT, TX, CO, NM, MT, AZ, WY
- Midwest: SD, OH, MO, AR, IN, MI, ND, MN
- Pacific west: CA, OR, NV, HI, WA, AS

Additionally, we have defined seasons as:
- Winter: Dec-Feb
- Spring: Mar-May
- Summer: Jun-Aug
- Fall: Sep-Nov

The metrics of the table are provided below as defined by the NPS on their "Visitor Use Statistics Definitions Page" <a name="ref5"></a>[5](#footnote5):

A visitor hour is defined by the NPS as "The presence of one or more persons, excluding NPS personnel, in a park for continuous, intermittent, or simultaneous periods of time aggregating one hour (e.g., one person for one hour, two persons for one-half hour, etc. Although same day attendance at detached parts of the same park are considered duplicate counts and not reported as separate visits, the time involved at detached units is added together for the visitor hour estimate." and becomes a recreation visitor hour when "associated with a recreation visit." All of the data we will be using will be related to recreation visits (as opposed to non-recreation visits that include brief pass throughs of park land, workers doing their jobs, and research activities).

A recreation visit is defined by the NPS as "The entry of a person onto lands or waters administered by the NPS except for non-reportable and non-recreation visits. Funeral parties at National Cemeteries, school groups, etc. are reportable as ‘recreation visits’ since their use aligns with the purpose for which the park was established. Visits originating on surface vehicles (trains, boats, other) and aircraft may be counted if they stop and disembark passengers on NPS administrated territory. The applicable rule is that one entrance per individual per day may be counted."


<a name="footnote5"></a>5. [^](#ref5) National Park Service. “NPS Visitor Use Statistics Definitions” *Www.nps.gov*, 21 Feb. 2024, https://www.nps.gov/subjects/socialscience/nps-visitor-use-statistics-definitions.htm


In [11]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE
## importing pandas for the data
import pandas as pd

## loading in the dataset
national_parks = pd.read_csv('data/00-raw/2000-2024 National Park Data.csv')

## Grabbing and renaming the columns that we are going to be analyzing
national_parks_cleaned = national_parks[['ParkName', 'Region', 'State', 'Year', 'Month', 'RecreationVisits', 'RecreationHours']]
national_parks_cleaned = national_parks_cleaned.rename(columns={
    "ParkName": "Park Name",
    "RecreationVisits": "Recreation Visits",
    "RecreationHours": "Recreation Hours"
})

## Changing the Visists # and Hours # into floats for calculation purpose for the Hours Per Visit Column
national_parks_cleaned["Recreation Visits"] = pd.to_numeric(
    national_parks_cleaned["Recreation Visits"].astype(str).str.replace(",", "")
)

national_parks_cleaned["Recreation Hours"] = pd.to_numeric(
    national_parks_cleaned["Recreation Hours"].astype(str).str.replace(",", ""),
)

## Adding the Hours Per Visit Column
national_parks_cleaned['Hours Per Visit'] = (national_parks_cleaned['Recreation Hours'] / national_parks_cleaned['Recreation Visits']).round(2)

## Size of Dataset
national_parks_cleaned.shape ## (14351, 8)

(14264, 8)

In [12]:
## Finding if there is missing values
national_parks_cleaned.isna().sum()

## Finding where the missing value exists
national_parks_cleaned[national_parks_cleaned.isna().any(axis=1)]

# 	Park Name	Region	State	Year	Month	Recreation Visits	Recreation Hours	Hours Per Visit
# 543	Arches NP	Intermountain	UT	2020	4	0	0	NaN
# 1143	Big Bend NP	Intermountain	TX	2020	4	0	0	NaN
# 1144	Big Bend NP	Intermountain	TX	2020	5	0	0	NaN
# 1146	Big Bend NP	Intermountain	TX	2020	7	0	0	NaN
# 2043	Canyonlands NP	Intermountain	UT	2020	4	0	0	NaN
# ...	...	...	...	...	...	...	...	...
# 13094	White Sands NP	Intermountain	NM	2020	4	0	0	NaN
# 13095	White Sands NP	Intermountain	NM	2020	5	0	0	NaN
# 13694	Yellowstone NP	Intermountain	WY	2020	4	0	0	NaN
# 13994	Yosemite NP	Pacific West	CA	2020	4	0	0	NaN
# 13995	Yosemite NP	Pacific West	CA	2020	5	0	0	NaN
# 84 rows × 8 columns

# These are the rows that includes N/A data. All of the data is for the Hours Per Visits column. This is not an error instead we can see a trend
# that all of these happens during 2020 which is the covid year, so the park most likely closed due to COVID restrictions. Therefore, there is
# 0 visits and 0 hours. Since Hours Per Visit is defined by Hours/Visits so showing NaN is expected. We are going to remove these rows as it 
# doesn't provide any insights for our analysis. Since this shows policy changes and not visitor behaviors. 

## Only keeping the rows that has recreation visists that is larger than 0
national_parks_cleaned = national_parks_cleaned[national_parks_cleaned["Recreation Visits"] > 0]

## New Size
national_parks_cleaned.shape # (14264, 8)

(14264, 8)

In [23]:
## Developing a IQR range to flag outliers (For the Recreation visits)
Q1 = national_parks_cleaned['Recreation Visits'].quantile(0.25)
Q3 = national_parks_cleaned['Recreation Visits'].quantile(0.75)

## Calculating the IQR
IQR = Q3 - Q1

## Constructing an upperbound
upper_bound = Q3 + 1.5 * IQR

## Flag outliers
visits_outliers = national_parks_cleaned[national_parks_cleaned['Recreation Visits'] > upper_bound]
## Showing the highest outliers
visits_outliers.sort_values('Recreation Visits', ascending = False)

Park Name          Region State  Year  Month  \
3905      Gateway Arch NP        Midwest     MO  2000      7   
3917      Gateway Arch NP        Midwest     MO  2001      7   
13709      Yellowstone NP  Intermountain     WY  2021      7   
3929      Gateway Arch NP        Midwest     MO  2002      7   
3965      Gateway Arch NP        Midwest     MO  2005      7   
...                   ...             ...   ...   ...    ...   
14059             Zion NP  Intermountain     UT  2000      9   
8706     Mount Rainier NP   Pacific West     WA  2000      8   
3378   Cuyahoga Valley NP        Midwest     OH  2006      8   
7191     Indiana Dunes NP        Midwest     IN  2024      5   
3461   Cuyahoga Valley NP        Midwest     OH  2013      7   

       Recreation Visits  Recreation Hours  Hours Per Visit  
3905             1162217           4648868             4.00  
3917             1160953           4643812             4.00  
13709            1080767          19645627            18.18  
3929             1064956           4259824             4.00  
3965             1021406           4085622             4.00  
...                  ...               ...              ...  
14059             289827           1975543             6.82  
8706              289775           5795500            20.00  
3378              289735            789316             2.72  
7191              289683            675510             2.33  
3461              289470            756156             2.61  

[1403 rows x 8 columns]

There are 1403 rows that are above the upper bound of the visits IQR, though this isn’t caused by errors. They are statistical outliers, because we can see that parks like Acadia and Zion are really popular and gain a lot of visitors. These observations reflect real-world park usage rather than data errors. Therefore, we will be keeping these values for the analysis.


In [22]:
## Constructing IQR for flagging outliers (For Hours Per Visits)
q1 = national_parks_cleaned['Hours Per Visit'].quantile(0.25)
q3 = national_parks_cleaned['Hours Per Visit'].quantile(0.75)
iqr = q3 - q1

## Calculating the upper bound
hpv_upper = q3 + 1.5 * iqr

## Finding the outliers
hpv_outliers = national_parks_cleaned[national_parks_cleaned['Hours Per Visit'] > hpv_upper]
## Showing the highest outliers
hpv_outliers.sort_values('Hours Per Visit', ascending = False)

Park Name          Region State  Year  Month  \
7280       Isle Royale NP        Midwest     MI  2006     10   
9827    North Cascades NP   Pacific West     WA  2023      1   
7383       Isle Royale NP        Midwest     MI  2015      5   
7279       Isle Royale NP        Midwest     MI  2006      9   
7291       Isle Royale NP        Midwest     MI  2007      9   
...                   ...             ...   ...   ...    ...   
13763         Yosemite NP   Pacific West     CA  2001      1   
2954   Channel Islands NP   Pacific West     CA  2021      4   
13762         Yosemite NP   Pacific West     CA  2000     12   
4550      Grand Canyon NP  Intermountain     AZ  2004      4   
8118   Lassen Volcanic NP   Pacific West     CA  2001      8   

       Recreation Visits  Recreation Hours  Hours Per Visit  
7280                  59             10416           176.54  
9827                   3               418           139.33  
7383                 947            114408           120.81  
7279                1070            127948           119.58  
7291                1265            129918           102.70  
...                  ...               ...              ...  
13763             102455           1464247            14.29  
2954               22512            321388            14.28  
13762             125999           1798295            14.27  
4550              409645           5844627            14.27  
8118               81285           1159437            14.26  

[1932 rows x 8 columns]

There are 1932 outliers according to the upper bound calculation using the IQR. We suspect that with this variable there might be more suspicious outliers. By looking at the highest few Hours Per Visit outliers we can observe some extreme numeric data. These cases are actually not data errors but reflect multiple day stays (small visitor count with long stays), or remote national parks (really far away) that require a longer stay. Since they are valid data we decided that it is appropriate to keep them in the analysis.

In [24]:
## Loading the Cleaned Data into the processed folder
processed_path = "data/02-processed/nps_visitation_2000_2024_processed.csv"
national_parks_cleaned.to_csv(processed_path, index = False)

In [25]:
## Loading the new and cleaned dataset
np = pd.read_csv("data/02-processed/nps_visitation_2000_2024_processed.csv")
## Fully Cleaned Dataset size
np.shape #(14264, 8)

np

Park Name          Region State  Year  Month  Recreation Visits  \
0      Acadia NP      Northeast     ME  2000      1              22500   
1      Acadia NP      Northeast     ME  2000      2              23848   
2      Acadia NP      Northeast     ME  2000      3              35423   
3      Acadia NP      Northeast     ME  2000      4              78555   
4      Acadia NP      Northeast     ME  2000      5             166456   
...          ...             ...   ...   ...    ...                ...   
14259    Zion NP  Intermountain     UT  2024      8             470995   
14260    Zion NP  Intermountain     UT  2024      9             506117   
14261    Zion NP  Intermountain     UT  2024     10             517501   
14262    Zion NP  Intermountain     UT  2024     11             311297   
14263    Zion NP  Intermountain     UT  2024     12             235404   

       Recreation Hours  Hours Per Visit  
0                148500             6.60  
1                157397             6.60  
2                233792             6.60  
3                455619             5.80  
4                965445             5.80  
...                 ...              ...  
14259           3221436             6.84  
14260           3153535             6.23  
14261           3338062             6.45  
14262           2035273             6.54  
14263           1453272             6.17  

[14264 rows x 8 columns]

In [34]:
## Summaryt statistics for each of the variables
national_parks_cleaned[['Recreation Visits', 'Recreation Hours', 'Hours Per Visit']].describe().round(2)

Recreation Visits  Recreation Hours  Hours Per Visit
count           14264.00          14264.00         14264.00
mean           108785.51         944642.10             7.65
std            149434.13        2055157.03             9.63
min                 2.00              4.00             0.15
25%             18595.75          70367.25             2.90
50%             54675.00         245063.00             4.66
75%            126848.75         762968.50             7.44
max           1162217.00       20212391.00           176.54

### Dataset #2 

See instructions above for Dataset #1.  Feel free to keep adding as many more datasets as you need.  Put each new dataset in its own section just like these. 

Lastly if you do have multiple datasets, add another section where you demonstrate how you will join, align, cross-reference or whatever to combine data from the different datasets

Please note that you can always keep adding more datasets in the future if these datasets you turn in for the checkpoint aren't sufficient.  The goal here is demonstrate that you can obtain and wrangle data.  You are not tied down to only use what you turn in right now.

In [ ]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE


## Ethics

Instructions: Keep the contents of this cell. For each item on the checklist
-  put an X there if you've considered the item
-  IF THE ITEM IS RELEVANT place a short paragraph after the checklist item discussing the issue.
  
Items on this checklist are meant to provoke discussion among good-faith actors who take their ethical responsibilities seriously. Your teams will document these discussions and decisions for posterity using this section.  You don't have to solve these problems, you just have to acknowledge any potential harm no matter how unlikely.

Here is a [list of real world examples](https://deon.drivendata.org/examples/) for each item in the checklist that can refer to.

[![Deon badge](https://img.shields.io/badge/ethics%20checklist-deon-brightgreen.svg?style=popout-square)](http://deon.drivendata.org/)

### A. Data Collection
 - [X] **A.1 Informed consent**: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data uses to which they consent?
 - [X] **A.2 Collection bias**: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?

> This project will use publicly available National Park Service (NPS) data, which is aggregated at the park level. A potential source of bias is that visitation counts may be influenced by factors like park accessibility, errors in recording, or park size. We will be careful not to interpret differences in visitation as reflecting individual visitor preferences.
 - [X] **A.3 Limit PII exposure**: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?
 - [X] **A.4 Downstream bias mitigation**: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

> A potential downstream risk is that results could be misinterpreted as reflecting individual preferences or behaviors. To mitigate this, we will frame our findings as park-level trends and avoid making claims about individual decision-making or protected groups.
### B. Data Storage
 - [X] **B.1 Data security**: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?
 - [X] **B.2 Right to be forgotten**: Do we have a mechanism through which an individual can request their personal information be removed?
 - [X] **B.3 Data retention plan**: Is there a schedule or plan to delete the data after it is no longer needed?

### C. Analysis
 - [X] **C.1 Missing perspectives**: Have we sought to address blindspots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?

> Our analysis focuses on visitation counts and camping participation and doesn't include qualitative visitor experiences, environmental impacts, or local community perspectives.
 - [X] **C.2 Dataset bias**: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?

> Our chosen dataset may reflect bias toward more popular or accessible parks as they tend to have more data collection systems. Smaller parks may have less accurate visitation estimates or records, which could affect regional comparisons.
 - [X] **C.3 Honest representation**: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?
 - [X] **C.4 Privacy in analysis**: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?
 - [X] **C.5 Auditability**: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?

### D. Modeling
 - [X] **D.1 Proxy discrimination**: Have we ensured that the model does not rely on variables or proxies for variables that are unfairly discriminatory?
 - [X] **D.2 Fairness across groups**: Have we tested model results for fairness with respect to different affected groups (e.g., tested for disparate error rates)?
 - [X] **D.3 Metric selection**: Have we considered the effects of optimizing for our defined metrics and considered additional metrics?

> We will use descriptive metrics such as visitation counts and recreation hours to examine park-level trends. We will also consider using derived measures such as average recreation hours per visit to provide context. These metrics are intended to describe patterns rather than evaluate performance or rank parks.
 - [X] **D.4 Explainability**: Can we explain in understandable terms a decision the model made in cases where a justification is needed?
 - [X] **D.5 Communicate limitations**: Have we communicated the shortcomings, limitations, and biases of the model to relevant stakeholders in ways that can be generally understood?

> We would specify that the metrics and variables that we are analyzing in our data isn't a metric that is used to measure park popularity or visitation satisfaction, and this results should not be interpreted as evidence of how environmental conditions affects recreational behaviors.

### E. Deployment
 - [X] **E.1 Monitoring and evaluation**: Do we have a clear plan to monitor the model and its impacts after it is deployed (e.g., performance monitoring, regular audit of sample predictions, human review of high-stakes decisions, reviewing downstream impacts of errors or low-confidence decisions, testing for concept drift)?

> Since this project will most likely become public after we are completed, we will monitor the results by looking through comments of people that looked through our project and feedback from other collaborators. We would be paying most of our attention on feedbacks that indicated we have misinterpreted certain findings and make changes to ensure that our analysis is accurate

 - [X] **E.2 Redress**: Have we discussed with our organization a plan for response if users are harmed by the results (e.g., how does the data science team evaluate these cases and update analysis and models to prevent future harm)?

> If we have users harmed by the results of our analysis we would quickly look into it as this matters a lot to us. We would correct our explanations, clairfy some limitation that we faced in our project, and revise our entire project and the parts that showed harm to users to ensure no harm is done.
 - [X] **E.3 Roll back**: Is there a way to turn off or roll back the model in production if necessary?

> If serious issues does happen, we would make the project private and revise everything that is necessary to ensure our project is accurate and non-harmful then we would make it public again.
 - [X] **E.4 Unintended use**: Have we taken steps to identify and prevent unintended uses and abuse of the model and do we have a plan to monitor these once the model is deployed?

> A potential unintended use of this project is treating our analysis (visitation counts or recreation hours) as indicators of vistor satisfaction, or environment quality. To prevent this kind of unintended use we would specify in our project that all of our analysis is descriptive and focuses on trends that connects to our variable, rather than saying which park is better or not or trying to take this as an evidence as how enviromental variables connects to recreational behaviors.


## Team Expectations 

Team Expectation 1: Work is evenly distributed 
Team Expectation 2: Respond to group messages within 12 hours
Team Expectation 3: Everything has to be done by 2pm on the day it is due
Team Expectation 4: If you know your are not able to do it notify in 24 hours in advanced
Team Expectation 5: Communication discord is what we are using
Team Expectation 6: Once a week in person on Tuesday 2pm
Team Expectation 7: Majority vote for big decisions
Team Expectation 8: Everyone can rotate the tasks
Team Expectation 9: if coding the same thing communciate with theother gorup or person that way their isn't conflicts and nothing gets lost
Team Expectation 10: Calm discussions, and don't be rude, as a way to handle group conflicts and most of all communicate


## Project Timeline Proposal

Instructions: Replace this with your timeline.  **PLEASE UPDATE your Timeline!** No battle plan survives contact with the enemy, so make sure we understand how your plans have changed.  Also if you have lost points on the previous checkpoint fix them

## Corrected/Modified

## Data (MODIFIED)

   
1. Explain what the **ideal** dataset you would want to answer this question. (This should include: 
   1. What variables?
      - Park name
      - Park region (NPS standard region classification)
      - Date (year + month)
      - Recreation visitations (monthly counts)
      - Average hours spent per visit (not just total visitor hours)
      - Seasons (dervired from month)
      - Park type (National Park)
      - Weather variables ( temperature, precipitations, snowfall)
      - Special events (free-entry days, major holidays, wildfire closures,etc)
   3. How many observations are needed?
      - Number of parks * 25 years * 12 months, gives thousands of oberservations, which is appropriate for seasonal and regional modeling
   5. Who/what/how would these data be collected?
      - National Park Service collects visitation using automated traffic counters at entrances
      - Parks without sensors reports counts manually
      - Visitor hours would ideally come from survyes or time-tracking studies
      - Weather data would coem from the nearest weather station
      - Special events would be recorded form NPS calendars and holidays
   7. How would these data be stored/organized?
      - data set should be tidy, analysis-ready table
      - each row is one park in one month
      - each column is one variable
2. Search for potential **real** datasets that could provide you with something useful for this project.  For each dataset that you find write 3-5 sentences describing 
   1. where the data is located (e.g., URL) and anything you need to do to use it (e.g., ask for permission, fill out an application)
  
   - https://irma.nps.gov/Stats/SSRSReports/National%20Reports/Query%20Builder%20for%20Public%20Use%20Statistics%20(1979%20-%20Last%20Calendar%20Year)
   - this is offical National Park Service visitaition database, no account, permission, or application required, can export monthly or annua data for any park
      
   2. what the important variables are in this dataset that you might use
      - recreation visits
      - Non-recreation visits
      - Visitor hours
      - Overnight stays
      - The dataset does not include everything that we need
         - the data set does not include Seasons we are defining it based on month
         - we don't have information on how many hours spent per visits
         - we also don't have any infomration in regards to the weather, if there were special events
  